src/test/java/core/basesyntax/service/OperationStrategyTest.java

```diff
@@ -0,0 +1,69 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.BalanceOperationHandler;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.PurchaseOperationHandler;
+import core.basesyntax.service.operation.ReturnOperationHandler;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class OperationStrategyTest {
+    private static final Map<FruitTransaction.Operation, OperationHandler> OPERATION_HANDLERS
+            = new HashMap<>();
+    private static final FruitTransaction.Operation BALANCE_OPERATION
+            = FruitTransaction.Operation.BALANCE;
+    private static final FruitTransaction.Operation PURSCHASE_OPERATION
+            = FruitTransaction.Operation.PURCHASE;
+    private static final FruitTransaction.Operation RETURN_OPERATION
+            = FruitTransaction.Operation.RETURN;
+    private static final FruitTransaction.Operation SUPPLE_OPERATION
+            = FruitTransaction.Operation.SUPPLY;
+    private static final OperationHandler BALANCE_OPERATION_HANDLER
+            = new BalanceOperationHandler();
+    private static final OperationHandler PURCHASE_OPERATION_HANDLER
+            = new PurchaseOperationHandler();
+    private static final OperationHandler RETURN_OPERATION_HANDLER
+            = new ReturnOperationHandler();
+    private OperationStrategy operationStrategy;
+
+    @BeforeEach
+    void setUp() {
+        operationStrategy = new OperationStrategy(OPERATION_HANDLERS);
+        OPERATION_HANDLERS.put(FruitTransaction.Operation.BALANCE, BALANCE_OPERATION_HANDLER);
+        OPERATION_HANDLERS.put(FruitTransaction.Operation.PURCHASE, PURCHASE_OPERATION_HANDLER);
+        OPERATION_HANDLERS.put(FruitTransaction.Operation.RETURN, RETURN_OPERATION_HANDLER);
+    }
+
+    @AfterEach
+    void afterAll() {
+        OPERATION_HANDLERS.clear();
+    }
```

Comment:
Do you really need a clear map after each test? 
you can initialize it in a static block or the BeforeAll method